lms/djangoapps/instructor/views/util_test_grades.py

```diff
@@ -0,0 +1,239 @@
+
+    def get_grades(self):
+        course_grades = []
+        for student in self.students:
+            course_grade_factory = CourseGradeFactory().create(student, self.course)
+            gradeset = course_grade_factory.summary
+            gradeset["username"] = course_grade_factory.user
+            course_grades.append(gradeset)
+
+        return course_grades
+
+    def by_section(self):
+        course_grade = self.get_grades()
+        score_by_section = []
+        header_rows = ['username', 'general_grade', 'fullname']
+        for student in course_grade:
+            course_grade_factory = CourseGradeFactory().create(student["username"], self.course)
+            sections = course_grade_factory.chapter_grades
+            course_data = CourseData(student, course=self.course)
+            course_policy = course_data.course.grading_policy
+
+            section_grade = DictList()
+            section_grade['username'] = student['username'].username
+            section_grade["fullname"] = student['username'].get_full_name()
+            section_grade["general_grade"] = student['percent']
+
+            for student_grade in student['section_breakdown']:
+                # In graders constructor, we added some additional keys
+                # in the section_breakdown json with the purpose to get the
+                # subsection's parent and be able to differentiate when a grade is
+                # calculated in a single entry. We apply this logic only if
+                # has a subsection object and discard the droppables.
+                if (student_grade.has_key('subsection') and
+                    student_grade['subsection'] is not None or
+                    student_grade.has_key('only') and not
+                    student_grade.has_key('mark')):
+
+                    locator = student_grade['subsection'].location
+                    parent = modulestore().get_parent_location(locator)
+                    parent_location = modulestore().get_item(parent)
+
+                    for policy in course_policy['GRADER']:
+                        if policy['type'] == student_grade['subsection'].format:
+                            grade = student_grade['percent'] * policy['weight']
+                            chapter_name = parent_location.display_name
+
+                            student_grade.update({'weight': policy['weight']})
+                            student_grade.update({'chapter_name': chapter_name})
+                            student_grade.update({'grade': grade})
+                            header_rows.append(chapter_name)
+
+                            # We group in a list the values that has the same keys.
+                            section_grade[student_grade['chapter_name']] = grade
+
+            for key, value in section_grade.items():
+                if isinstance(value, (list,)):
+                    qty = len(value)
+                    value = sum(value)
+                    result = value/qty
```

Comment:
@diegomillan We already know that this is not fine. We need to average according to the number of assignment types in all course.